processors/pubmed_unregistered_trials/processor.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import logging
+from sqlalchemy import exc as sqlalchemy_exceptions
+from .. import base
+from . import extractors as extractors_module
+from ..pubmed_publications import extractors as related_extractors_module
+logger = logging.getLogger(__name__)
+
+
+def process(conf, conn):
+    extractors = base.helpers.get_variables(
+        extractors_module, lambda x: x.startswith('extract_')
+    )
+    related_extractors = base.helpers.get_variables(
+        related_extractors_module, lambda x: x.startswith('extract_')
+    )
+    source = related_extractors['extract_source'](None)
+    source_id = base.writers.write_source(conn, source)
+
+    for record in base.helpers.iter_rows(conn, 'warehouse', 'pubmed', orderby='meta_id'):
+        publication = related_extractors['extract_publication'](record)
+        unregistered_trial = extractors['extract_trial'](record)
+        conn['database'].begin()
+        try:
+
+            # Create/Update unregistered trial from the publication
+            if not publication['registry_ids']:
+                trial_id, is_primary = base.writers.write_trial(
+                    conn, unregistered_trial, source_id, record['meta_id']
+                )
+                base.writers.write_record(
+                    conn, record, source_id, trial_id, unregistered_trial, is_primary
+                )
+
+            # Publication is not/no longer an unregistered trial but a results document
+            else:
+                delete_record = "DELETE FROM records WHERE identifiers @> '%s'"
+                delete_record = delete_record % json.dumps(unregistered_trial['identifiers'])
+                conn['database'].query(delete_record)
+                for identifiers in publication['registry_ids']:
+                    trial = base.helpers.find_trial_by_identifiers(
+                        conn, identifiers=identifiers
+                    )
+                    if trial:
+                        doc_category_id = extractors['extract_document_category'](record)
+                        document = extractors['extract_document'](record)
+                        document.update({
+                            'source_id': source_id,
+                            'document_category_id': doc_category_id,
+                        })
```

Comment:
As for `extract_document` I would like to keep the convention but I don't agree with it. Even though the function is called `extract_documents` it extracts only 1 document in all processors that have it. 
I think we should rename the function `extract_document` in all other processors and avoid doing [this](https://github.com/opentrials/processors/blob/028150961ce1b5686ec9040201f9bb5e710a8774/processors/gsk/extractors.py#L128). What do you think?